### PR TITLE
[SPARK-47982][BUILD] Update some code style's plugins to latest version

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to update some some code style's plugins to latest version, include:
- `mvn-scalafmt_2.13` from `1.1.1684076452.9f83818` to `1.1.1713302731.c3d0074`.
- `checkstyle` from `10.14.0` to `10.15.0`.
- `scalafmt` from `3.8.0` to `3.8.1`.

### Why are the changes needed?
1.`mvn-scalafmt_2.13`
https://github.com/SimonJPegg/mvn_scalafmt/releases/tag/v2.13-1.1.1713302731.c3d0074
Minor version bumps and dropping 2.12

2.`checkstyle`
https://checkstyle.org/releasenotes.html#Release_10.15.0
https://checkstyle.org/releasenotes.html#Release_10.14.2
https://checkstyle.org/releasenotes.html#Release_10.14.1

3.`scalafmt`
https://github.com/scalameta/scalafmt/releases/tag/v3.8.1
image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by panbingkun** [comment]
@HyukjinKwon 
So quickly approved, thank you very much! ❤️

File: project/plugins.sbt [hunk 1]
```diff
@@ -20,10 +20,10 @@ addSbtPlugin("software.purpledragon" % "sbt-checkstyle-plugin" % "4.0.1")
 // sbt-checkstyle-plugin uses an old version of checkstyle. Match it to Maven's.
 // If you are changing the dependency setting for checkstyle plugin,
 // please check pom.xml in the root of the source tree too.
-libraryDependencies += "com.puppycrawl.tools" % "checkstyle" % "10.14.0"
+libraryDependencies += "com.puppycrawl.tools" % "checkstyle" % "10.15.0"
 
-// checkstyle uses guava 31.0.1-jre.
-libraryDependencies += "com.google.guava" % "guava" % "31.0.1-jre"
+// checkstyle uses guava 33.1.0-jre.
+libraryDependencies += "com.google.guava" % "guava" % "33.1.0-jre"
```
Comment (dongjoon-hyun) [inline project/plugins.sbt:26]: Oh, which `checkstyle` version upgrade this?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you.

File: project/plugins.sbt (hunk 1 above)
Comment (panbingkun) [reply project/plugins.sbt:26]: the version `10.15.0` use it. I have check it.
https://github.com/checkstyle/checkstyle/blob/checkstyle-10.15.0/pom.xml#L289-L292
image

**Comment by yaooqinn** [comment]
Merged to master. Thank you @panbingkun @HyukjinKwon @dongjoon-hyun

**Pull request closed without merging**